common.sh

```diff
@@ -35,9 +35,30 @@ delete_pg()
     if [ ${ENABLE_PLACEMENT_GROUP} -eq 0 ] || [ -z $PLACEMENT_GROUP ]; then
         return 0
     fi
-    AWS_DEFAULT_REGION=us-west-2 aws ec2 delete-placement-group \
-        --group-name ${PLACEMENT_GROUP}
-    return $?
+    local ret=0
+    local retry=20
+    while [ "$retry" -ge 0 ]; do
+        echo "Start deleting placement group ${PLACEMENT_GROUP}."
+        delete_pg_response=$(AWS_DEFAULT_REGION=us-west-2 aws ec2 delete-placement-group \
+            --group-name ${PLACEMENT_GROUP} 2>&1)
+        ret=$?
+        # The placement group may be in use due to the attached
+        # ec2 instances are not terminated completely. Keep
+        # waiting and retrying within 20*30=600 seconds (10 minutes).
+        if [[ "$ret" -ne 0 && "$delete_pg_response" == *"InvalidPlacementGroup.InUse"* ]]; then
+            echo "The placement group ${PLACEMENT_GROUP} is in use and may not be deleted, retry after 30 seconds."
```

Comment:
Similar to previous patch, I don't think we need a bunch of prints in the loop.

Reply:
Fixed in latest revision.